src/ImageSharp/Common/Memory/BufferSpan.cs

```diff
@@ -16,59 +15,48 @@ namespace ImageSharp
     internal static class BufferSpan
     {
         /// <summary>
-        /// It's worth to use Marshal.Copy() or Buffer.BlockCopy() over this size.
-        /// </summary>
-        private const int ByteCountThreshold = 1024;
-
-        /// <summary>
         /// Copy 'count' number of elements of the same type from 'source' to 'dest'
         /// </summary>
         /// <typeparam name="T">The element type.</typeparam>
-        /// <param name="source">The input <see cref="BufferSpan{T}"/></param>
+        /// <param name="source">The <see cref="BufferSpan{T}"/> to copy elements from.</param>
         /// <param name="destination">The destination <see cref="BufferSpan{T}"/>.</param>
         /// <param name="count">The number of elements to copy</param>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static void Copy<T>(BufferSpan<T> source, BufferSpan<T> destination, int count)
+        public static unsafe void Copy<T>(BufferSpan<T> source, BufferSpan<T> destination, int count)
             where T : struct
         {
-            CopyImpl(source, destination, count);
-        }
+            DebugGuard.MustBeLessThanOrEqualTo(count, source.Length, nameof(count));
+            DebugGuard.MustBeLessThanOrEqualTo(count, destination.Length, nameof(count));
 
-        /// <summary>
-        /// Copy 'countInSource' elements of <typeparamref name="T"/> from 'source' into the raw byte buffer 'destination'.
-        /// </summary>
-        /// <typeparam name="T">The element type.</typeparam>
-        /// <param name="source">The source buffer of <typeparamref name="T"/> elements to copy from.</param>
-        /// <param name="destination">The destination buffer.</param>
-        /// <param name="countInSource">The number of elements to copy from 'source'</param>
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static void Copy<T>(BufferSpan<T> source, BufferSpan<byte> destination, int countInSource)
-            where T : struct
-        {
-            CopyImpl(source, destination, countInSource);
+            ref byte srcRef = ref Unsafe.As<T, byte>(ref source.DangerousGetPinnableReference());
+            ref byte destRef = ref Unsafe.As<T, byte>(ref destination.DangerousGetPinnableReference());
+
+            int byteCount = Unsafe.SizeOf<T>() * count;
+
+            // TODO: Use unfixed Unsafe.CopyBlock(ref T, ref T, int) for small blocks, when it gets available!
+            fixed (byte* pSrc = &srcRef)
+            fixed (byte* pDest = &destRef)
+            {
+#if NETSTANDARD1_1
+                Unsafe.CopyBlock(pDest, pSrc, (uint)byteCount);
```

Comment:
@JimBobSquarePants this line is uncovered.
Currently we are not able to test code on `#if NETSTANDARD1_1` path. We should change our test project configuration, or drop the 1.1 target.